dsa.go

```diff
@@ -23,22 +23,32 @@ func (k Key) readDSA() (*dsa.PrivateKey, error) {
 		return nil, fmt.Errorf("invalid header inside public key: %q: expected %q", pub.Header, k.Algo)
 	}
 
+	publicKey := &dsa.PublicKey{
+		Parameters: dsa.Parameters{
+			P: pub.P,
+			Q: pub.Q,
+			G: pub.G,
+		},
```

Comment:
The Y member is missed.

```suggestion
		},
		Y: pub.Pub,
```